Common/src/main/java/com/blamejared/crafttweaker/impl/plugin/crafttweaker/ZenClassRegistrationManager.java

```diff
@@ -0,0 +1,152 @@
+package com.blamejared.crafttweaker.impl.plugin.crafttweaker;
+
+import com.blamejared.crafttweaker.api.annotation.Preprocessor;
+import com.blamejared.crafttweaker.api.natives.NativeTypeInfo;
+import com.blamejared.crafttweaker.api.plugin.IJavaNativeIntegrationRegistrationHandler;
+import com.blamejared.crafttweaker.api.util.InstantiationUtil;
+import com.blamejared.crafttweaker.api.zencode.IPreprocessor;
+import com.blamejared.crafttweaker.api.zencode.ZenTypeInfo;
+import com.blamejared.crafttweaker_annotations.annotations.NativeMethod;
+import com.blamejared.crafttweaker_annotations.annotations.NativeTypeRegistration;
+import com.blamejared.crafttweaker_annotations.annotations.TypedExpansion;
+import org.openzen.zencode.java.ZenCodeType;
+
+import java.util.ArrayDeque;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Queue;
+
+final class ZenClassRegistrationManager {
+    
+    @FunctionalInterface
+    private interface LateCallback {
+        
+        void callback(final Class<?> clazz, final String loader, final IJavaNativeIntegrationRegistrationHandler handler);
+        
+    }
+    
+    private record LateRegistrationCandidate(Class<?> clazz, String loader, LateCallback callback) {}
```

Comment:
Future proofing in case we need late registration for something else; the impact it has on complexity is arguably negligible, so I figured I might as well